profiles/schema.py

```diff
@@ -172,6 +176,62 @@ def resolve_total_count(self, info, **kwargs):
         return self.iterable.model.objects.count()
 
 
+class PrimaryContactInfoOrderingFilter(OrderingFilter):
+    """Custom ordering filter
+
+    This filter enables ordering profiles by primary contact info fields, for example by city of primary address.
+    """
+
+    # custom field definitions:
+    # 0. custom field name (camel case format)
+    # 1. field display text
+    # 2. model with foreign key profile_id
+    # 3. field name of the related model
+
+    FIELDS = (
+        ("primaryCity", "Primary City", Address, "city"),
+        ("primaryPostalCode", "Primary Postal Code", Address, "postal_code"),
+        ("primaryAddress", "Primary Address", Address, "address"),
+        ("primaryCountryCode", "Primary Country Code", Address, "country_code"),
+        ("primaryEmail", "Primary Email", Email, "email"),
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # add ascending and descending custom orderings
+        self.extra["choices"] += [(item[0], item[1]) for item in self.FIELDS]
+        self.extra["choices"] += [
+            (f"-{item[0]}", f"{item[1]} (Descending)") for item in self.FIELDS
+        ]
+
+    def filter(self, qs, values):
+        def _get_field_data_by_field(fields, field):
+            for f in fields:
+                if f[0] == field:
+                    return f
+            return None
+
+        # collect all the ascending/descending options and flatten the list
+        options = chain(*[[item[0], f"-{item[0]}"] for item in self.FIELDS])
+
+        for value in values or []:
+            # match with all of our custom ascending and descending orderings
+            if value in options:
+                # get rid of leading "-" if ordering is descending
+                field_name = value.replace("-", "")
+                field_data = _get_field_data_by_field(self.FIELDS, field_name)
+                # annotate the query with extra field and return the queryset
+                annotation = {
+                    to_snake_case(field_name): Subquery(
+                        field_data[2]
```

Comment:
so, if `field_data == None` you get nice exceptions here? 🙂

Reply:
Yep. I was just iterating the idea here hence the draft pr. I'm still going to make some checks there and actually tests alone will keep malicious configurations out of the production.

Reply:
but yeah, it should not get here ever, since you check `if value in options`...

Reply:
Yeah. I guess I'm still going to do the absolutely necessary checks to the final pr but the tests alone will catch most of the toxic configuration 😃